api/core/tools/provider/builtin/searchapi/searchapi.py

```diff
@@ -0,0 +1,29 @@
+from typing import Any
+
+from core.tools.entities.values import ToolLabelEnum
+from core.tools.errors import ToolProviderCredentialValidationError
+from core.tools.provider.builtin.searchapi.tools.google import GoogleTool
+from core.tools.provider.builtin_tool_provider import BuiltinToolProviderController
+
+
+class SearchAPIProvider(BuiltinToolProviderController):
+    def _validate_credentials(self, credentials: dict[str, Any]) -> None:
+        try:
+            GoogleTool().fork_tool_runtime(
+                runtime={
+                    "credentials": credentials,
+                }
+            ).invoke(
+                user_id='',
+                tool_parameters={
+                    "query": "SearchApi dify",
+                    "result_type": "link"
+                },
+            )
+        except Exception as e:
+            raise ToolProviderCredentialValidationError(str(e))
+
+    def _get_tool_labels(self) -> list[ToolLabelEnum]:
```

Comment:
I talked to @Yeuoly yesterday. This is no longer needed. We move those tags to the yaml file.

https://github.com/langgenius/dify/blob/74f38eacdafb777ac68197333648ffbff0dcc7c9/api/core/tools/provider/builtin/bing/bing.yaml#L14